src/arrayMethodSort.js

```
@@ -4,8 +4,25 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
```

If `compareFunction` is provided but is not a function (e.g., a string or number), the native `Array.prototype.sort` throws a `TypeError`. Your implementation does not check for this and will silently use the default comparator. To comply with the requirements, you must add a check at the start of the function:

```javascript
if (compareFunction !== undefined && typeof compareFunction !== 'function') {
  throw new TypeError('compareFunction must be a function');
}
```

This ensures your method matches the native sort's error handling.